mono/eglib/gstr.c

```diff
@@ -886,6 +886,34 @@ g_ascii_strcasecmp (const gchar *s1, const gchar *s2)
 	return (*sp1) - (*sp2);
 }
 
+gboolean
+g_utf16_ascii_equal (const gunichar2 *utf16, size_t ulen, const char *ascii, size_t alen)
+{
+	size_t i;
+	if (ulen != alen)
+		return FALSE;
+	for (i = 0; i < ulen; ++i) {
+		if (utf16[i] != ascii[i])
+			return FALSE;
+	}
+	return TRUE;
+}
+
+gboolean
+g_utf16_asciiz_equal (const gunichar2 *utf16, const char *ascii)
+// z for zero means null terminated
+{
+	while (1)
+	{
+		char a = *ascii++;
+		gunichar2 u = *utf16++;
```

Comment:
Note these functions are wonky for "ascii" in the range 80-FF, if you consider that ascii.

Reply:
typically char 0x80 will not compare equal to unichar2 0x80 -- on most but maybe not all systems.